src/client.rs

```diff
@@ -1138,6 +1194,17 @@ impl Client {
             "uploads/",
         )
     }
+
+    fn to_list_tags_url(&self, reference: &Reference) -> String {
+        format!(
+            "{}://{}/v2/{}/tags/list",
+            self.config
+                .protocol
+                .scheme_for(reference.resolve_registry()),
+            reference.resolve_registry(),
+            reference.repository(),
+        )
```

Comment:
nitpicking, I'm a bit paranoid and I would prefer to have the url built using a library that can handle escaping and validate also the final url.

I suggest to look into something reqwest's [`Url::parse_with_params`](https://docs.rs/reqwest/0.11.17/reqwest/struct.Url.html#method.parse_with_params).

Reply:
[according to spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pulling-manifests) there can't be any characters in the `name`.

should i wrap it anyway?

Reply:
Good, let's stay with `format!` then